toolchain/check/testdata/interop/cpp/typedef.carbon

```diff
@@ -0,0 +1,235 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/int.carbon
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interop/cpp/typedef.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/interop/cpp/typedef.carbon
+
+// --- i32_typedef.h
+
+typedef int foo;
+
+using bar = int;
+
+// --- use_i32_typedef.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "i32_typedef.h";
+
+fn F() {
+  //@dump-sem-ir-begin
+  var n: Cpp.foo = 42;
+  // Types are equivalent.
+  var p: Cpp.bar* = &n;
+  //@dump-sem-ir-end
+}
+
+// --- class_typedef.h
+
+class C {};
+using D = C;
```

Comment:
Shouldn't this be a `typedef` instead of `using`? I'd have thought `using` would look a little different in the AST. May also be good to verify `using` works, but it didn't seem like the focus of this support...
```suggestion
typedef C D;
```

Reply:
They're really just different syntaxes for the same thing (in the clang AST they're two different derived classes of the same `TypedefNameDecl` base class just to preserve the spelling, but they behave the same). The `i32` test has examples using both syntaxes; so the other syntax is covered, but outside of the test for that particular syntax I'd prefer to use the more modern `using` syntax.